code/gerasimov_dmitry_lab1/solving_quadratic_equation.py

```diff
@@ -0,0 +1,50 @@
+import math
+
+
+def is_correct_input_type(value):
+    if type(value) is float or type(value) is int:
+        return True
+    return False
+
+
+class QuadraticEquation:
+    def __init__(self, a=1, b=0, c=0):
+        if not is_correct_input_type(a) or not is_correct_input_type(b) or not is_correct_input_type(c):
+            raise TypeError("The arguments type must be 'float' or 'int'")
+        if a == 0:
+            raise ValueError("The equation is not square. The first argument cannot be zero")
+        self.answer = None
+        self.a = a
+        self.b = b
+        self.c = c
+
+    def solution(self):
+        self.answer = list()
+        disc = self.b ** 2 - 4 * self.a * self.c
+        if self.b == 0:
+            if self.c == 0:
+                disc = 0
```

Comment:
Looks like if `b == 0 and c == 0`, then` disc = self.b ** 2 - 4 * self.a * self.c = 0`